From a5903891f090b475a9d636fdb35cf06867c1ced4 Mon Sep 17 00:00:00 2001
From: Paul Lesniewski <paul@squirrelmail.org>
Date: Thu, 19 Aug 2010 00:06:36 +0000
Subject: [PATCH 04/50] Add warning about using server_port as 993 (imaps isn't
 supported) to log output so it's harder to miss such mistakes

git-svn-id: https://svn.code.sf.net/p/squirrelmail/code/trunk/imap_proxy@14050 7612ce4b-ef26-0410-bec9-ea0150e637f0
---
 src/main.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/src/main.c b/src/main.c
index dfd3113..757d0ac 100644
--- a/src/main.c
+++ b/src/main.c
@@ -813,8 +813,16 @@ static void ServerInit( void )
     }
 
     syslog(LOG_INFO, "%s: Proxying to IMAP port %s", 
-	   fn, PC_Struct.server_port );
+           fn, PC_Struct.server_port );
         
+    // Since IMAP Proxy uses STARTTLS and does not speak "imaps",
+    // let's spit out an informative warning message if the server
+    // port was configured to 993, which, 99.9% of the time, is a
+    // mistake
+    //
+    if (strcmp(PC_Struct.server_port, "993") == 0)
+        syslog(LOG_ERR, "WARNING: IMAP Proxy uses STARTTLS to encrypt a \"normal\" IMAP connection and does not support direct TLS/SSL connections that are typically served on port 993.  Chances are you have misconfigured the server_port setting, and that it should be something more like port 143.  If the server at '%s' supports STARTTLS from an unencrypted connection on port 993, then you can ignore this (but, again, chances are that this is NOT the case).", PC_Struct.server_hostname);
+
     /* 
      * fill in the address family, the host address, and the
      * service port of our global socket address structure
-- 
2.7.4

